### PR TITLE
fix: [PIE-4290]: Fix gitsync popup opening for inline template

### PR DESCRIPTION
**Pull request by depanshu-dhiman**
### Summary
gitDetails were being sent for new git experience. Prevented that and used storeMetadata for the git values 

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

## [Contributor license agreement](https://github.com/harness/harness-core-ui/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 1 Issues 
* Code issue 1 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/10970)

File: src/modules/10-common/components/SaveToGitFormV2/__tests__/SaveToGitFormV2.test.tsx [hunk 1]
```diff
@@ -113,4 +113,56 @@ describe('SaveToGitFormV2 test', () => {
...
+            }
+          }}
+          onSuccess={onSuccess}
+          {...pathParams}
+        />
+      </TestWrapper>
+    )
+
+    const label = getByText('common.git.existingBranchCommitLabel')
+    const branch = label.nextElementSibling as HTMLSpanElement
```
Comment (chetanvekariya-harness) [inline src/modules/10-common/components/SaveToGitFormV2/__tests__/SaveToGitFormV2.test.tsx:144]: This could break easily. Rather use getByText('branch1')

**Review by sonaligoyal-harness — DISMISSED** [review]

File: src/modules/10-common/components/SaveToGitFormV2/__tests__/SaveToGitFormV2.test.tsx [hunk 2]
```diff
@@ -113,4 +113,56 @@ describe('SaveToGitFormV2 test', () => {
...
+    expect(getByText('common.git.existingBranchCommitLabel')).toBeInTheDocument()
```
Comment (chetanvekariya-harness) [inline src/modules/10-common/components/SaveToGitFormV2/__tests__/SaveToGitFormV2.test.tsx:145]: Use variable label that you created above.

File: src/modules/10-common/components/SaveToGitFormV2/__tests__/SaveToGitFormV2.test.tsx [hunk 3]
```diff
@@ -113,4 +113,56 @@ describe('SaveToGitFormV2 test', () => {
...
+    expect(branch.innerHTML).toBe('branch1')
+
+    const labelNewBranch = getByText('common.git.newBranchCommitLabel')
+    fireEvent.click(labelNewBranch)
+
+    const input = container.querySelector('input[name=branch]') as HTMLInputElement
```
Comment (chetanvekariya-harness) [inline src/modules/10-common/components/SaveToGitFormV2/__tests__/SaveToGitFormV2.test.tsx:151]: Use **queryByNameAttribute**

File: src/modules/10-common/components/SaveToGitFormV2/__tests__/SaveToGitFormV2.test.tsx [hunk 4]
```diff
@@ -113,4 +113,56 @@ describe('SaveToGitFormV2 test', () => {
...
+          onSuccess={onSuccess}
+          {...pathParams}
+        />
+      </TestWrapper>
+    )
+
+    const label = getByText('common.git.existingBranchCommitLabel')
+    const branch = label.nextElementSibling as HTMLSpanElement
+    expect(getByText('common.git.existingBranchCommitLabel')).toBeInTheDocument()
+    expect(branch.innerHTML).toBe('branch1')
```
Comment (chetanvekariya-harness) [inline src/modules/10-common/components/SaveToGitFormV2/__tests__/SaveToGitFormV2.test.tsx:146]: If you use **getByText('branch1')** then you do not need this.

**Review by chetanvekariya-harness — DISMISSED** [review]

File: src/modules/10-common/components/SaveToGitFormV2/__tests__/SaveToGitFormV2.test.tsx (hunk 4 above)
Comment (depanshu-dhiman) [reply src/modules/10-common/components/SaveToGitFormV2/__tests__/SaveToGitFormV2.test.tsx:146]: done

File: src/modules/10-common/components/SaveToGitFormV2/__tests__/SaveToGitFormV2.test.tsx (hunk 3 above)
Comment (depanshu-dhiman) [reply src/modules/10-common/components/SaveToGitFormV2/__tests__/SaveToGitFormV2.test.tsx:151]: done

File: src/modules/10-common/components/SaveToGitFormV2/__tests__/SaveToGitFormV2.test.tsx (hunk 2 above)
Comment (depanshu-dhiman) [reply src/modules/10-common/components/SaveToGitFormV2/__tests__/SaveToGitFormV2.test.tsx:145]: done

File: src/modules/10-common/components/SaveToGitFormV2/__tests__/SaveToGitFormV2.test.tsx (hunk 1 above)
Comment (depanshu-dhiman) [reply src/modules/10-common/components/SaveToGitFormV2/__tests__/SaveToGitFormV2.test.tsx:144]: done

**Review by YounixIO — APPROVED** [review]

**Review by abhinavrastogi-harness — APPROVED** [review]

**Pull request merged**